pkg/minikube/tunnel/reporter.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tunnel
+
+import (
+	"fmt"
+
+	"io"
+	"k8s.io/minikube/pkg/minikube/tunnel/types"
+	"reflect"
+	"strings"
+)
+
+type Reporter interface {
+	Report(tunnelState *types.TunnelState)
+}
+
+type SimpleReporter struct {
+	out       io.Writer
+	lastState *types.TunnelState
+}
+
+func (r *SimpleReporter) Report(tunnelState *types.TunnelState) {
+	if reflect.DeepEqual(r.lastState, tunnelState) {
```

Comment:
reflect is generally frowned upon outside of testing code. You should be able to use simple equality here on the struct values.

Reply:
fixed